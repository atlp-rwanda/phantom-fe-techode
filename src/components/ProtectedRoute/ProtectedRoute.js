import React from "react";
import { Route, Redirect } from "react-router-dom";

const ProtectedRoute = ({ auth, component: Component, ...rest }) => {
  let logginStatus = localStorage.getItem("Logged");
  return (
    <Route
      {...rest}
      render={(props) => {
        if (logginStatus) {
          return <Component {...props} />;
        } else {
          return (
            <Redirect to={{ path: "/", state: { from: props.location } }} />
          );
        }
     
      }}
    />
  );
};

export default ProtectedRoute;
