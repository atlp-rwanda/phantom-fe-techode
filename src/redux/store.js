import { applyMiddleware, createStore } from "redux";
import reducers from './reducers/index';
import reduxThunk from 'redux-thunk'


export const middlewares = [ reduxThunk ];
export const createStoreWithMiddleware = applyMiddleware(...middlewares)(createStore);

/* ========= Start: configuration store ========= */ 
const  store = createStoreWithMiddleware(reducers , {} , window.__REDUX_DEVTOOLS_EXTENSION__ && window.__REDUX_DEVTOOLS_EXTENSION__())
/* ========== End: configuration store ========== */ 

export default store