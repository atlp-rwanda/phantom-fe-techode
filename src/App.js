import React, { useEffect } from "react";
import {
  BrowserRouter as Router,
  Redirect,
  Route,
  Switch,
} from "react-router-dom";
import { connect } from "react-redux";

import Login from "./containers/SignIn/LoginForm.js";
import Home from "./containers/Home/Home.js";
import Logout from "./components/logout/Logout.js";
import Profile from "./containers/profile/Profile.js";
import NewPassword from "./components/passwordReset/newpassword.js";
import RegisterDriver from "./containers/RegisterDriverOperator/RegisterDriver.js";
import RegisterOperator from "./containers/RegisterDriverOperator/RegisterOperator.js";
import RegisterBus from "./containers/RegisterBus/RegisterBus.js";
import "react-toastify/dist/ReactToastify.css";
import "./assets/style/style-skeleton.css";
import Dashboard from "./containers/Dashboard/Dashboard.js";
import ProtectedRoute from "./components/ProtectedRoute/ProtectedRoute.js";
import Roles from "./components/roles/Roles.js";
import Busesoperat from "./containers/busesOp/Busesoperat.js";
import BusSimulation from "./containers/BusSimulation/BusSimulation.js";
import OperatorAssignBus from "./containers/OperatorDriverBus/OperatorAssignBus.js";
import AddRoute from "./containers/Routes/AddRoute.js";
import SignUp from "./containers/SignUp/SignUp.js";
import PasswordReset from "./components/passwordReset/passwordereset.js";
import Simulation from "./containers/Simulation/Simulation.js";

const App = (props) => {
  const authentication = props.authentication;

  let logginStatus = localStorage.getItem("Logged");

  return (
    <Router>
      <Switch>
        <Route
          exact
          path="/"
          render={() =>
            logginStatus ? <Redirect to="/dashboard" /> : <Home />
          }
        />
         <Route
          exact
          path="/tracking"
          render={() =>
            <Simulation />
          }
        />

        <Route
          path="/reset"
          render={() =>
            logginStatus ? <Redirect to="/dashboard" /> : <PasswordReset />
          }
        />
        <Route
          path={`/newpassword/:${props.token}`}
          render={() =>
            logginStatus ? <Redirect to="/dashboard" /> : <NewPassword />
          }
        />
        
        <Route
          path="/login"
          render={() =>
            logginStatus ? <Redirect to="/dashboard" /> : <Login />
          }
        />

        <Route
          path="/signup"
          render={() =>
            logginStatus ? <Redirect to="/dashboard" /> : <SignUp />
          }
        />

        <ProtectedRoute
          path="/dashboard"
          component={Dashboard}
          auth={authentication.auth}
        />
        <ProtectedRoute
          path="/routes"
          component={AddRoute}
          auth={authentication.auth}
        />
        <ProtectedRoute
          path="/drivers"
          component={RegisterDriver}
          auth={authentication.auth}
        />
        <ProtectedRoute
          path="/operators"
          component={RegisterOperator}
          auth={authentication.auth}
        />
        <ProtectedRoute
          path="/profile"
          component={Profile}
          auth={authentication.auth}
        />
        <ProtectedRoute
          path="/roles"
          component={Roles}
          auth={authentication.auth}
        />
        <ProtectedRoute
          path="/buses"
          component={Busesoperat}
          auth={authentication.auth}
        />
        <ProtectedRoute
          path="/simulation"
          component={BusSimulation}
          auth={authentication.auth}
        />
        <ProtectedRoute
          path="/buses"
          component={RegisterBus}
          auth={authentication.auth}
        />
        <ProtectedRoute
          path="/routes"
          component={AddRoute}
          auth={authentication.auth}
        />
        <ProtectedRoute
          path="/dashboard_operator"
          component={Dashboard}
          auth={authentication.auth}
        />
        <ProtectedRoute
          path="/assign_drivers_buses"
          component={OperatorAssignBus}
          auth={authentication.auth}
        />
        <Route path="/*" component={() => ({ NotFound: 400 })} />
      </Switch>
    </Router>
  );
};

const mapToState = (state) => {
  return {
    authentication: state.authentication,
  };
};

export default connect(mapToState, {})(App);
