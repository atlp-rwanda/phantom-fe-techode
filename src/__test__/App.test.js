import React from 'react';
import { shallow  } from 'enzyme';
import toJson from 'enzyme-to-json';

import store from "../redux/store"
import App from '../App';
import { BrowserRouter as Router, Route, Switch } from 'react-router-dom';
import ProtectedRoute from '../components/ProtectedRoute/ProtectedRoute';

describe('<App />', () =>{
  
        let wrapper;
      beforeEach(()=>{
        wrapper = shallow (<App store={store} /> ).childAt(0).dive();
      });
  
      it('It should match the snapshot', () => { 
            expect(toJson(wrapper)).toMatchSnapshot();
      });
    
      it('if it should render one Siwtch', () =>{
        expect(wrapper.find(Switch).length).toBe(1);
      });
  
      it('if it should render one Route', () =>{
          expect(wrapper.find(Route).length).toBe(7);
        });
      
      it('if it should render ProtectedRoutes', () =>{
        expect(wrapper.find(ProtectedRoute).length).toBe(12);
      });
  
      it('if it should render one Router', () =>{
        expect(wrapper.find(Router).length).toBe(1);
      });

    });