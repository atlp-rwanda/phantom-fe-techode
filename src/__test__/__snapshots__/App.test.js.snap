// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App /> It should match the snapshot 1`] = `
<BrowserRouter>
  <Switch>
    <Route
      exact={true}
      path="/"
      render={[Function]}
    />
    <Route
      exact={true}
      path="/tracking"
      render={[Function]}
    />
    <Route
      path="/reset"
      render={[Function]}
    />
    <Route
      path="/newpassword/:undefined"
      render={[Function]}
    />
    <Route
      path="/login"
      render={[Function]}
    />
    <Route
      path="/signup"
      render={[Function]}
    />
    <ProtectedRoute
      auth={false}
      component={
        Object {
          "$$typeof": Symbol(react.memo),
          "WrappedComponent": [Function],
          "compare": null,
          "type": [Function],
        }
      }
      path="/dashboard"
    />
    <ProtectedRoute
      auth={false}
      component={
        Object {
          "$$typeof": Symbol(react.memo),
          "WrappedComponent": [Function],
          "compare": null,
          "type": [Function],
        }
      }
      path="/routes"
    />
    <ProtectedRoute
      auth={false}
      component={[Function]}
      path="/drivers"
    />
    <ProtectedRoute
      auth={false}
      component={[Function]}
      path="/operators"
    />
    <ProtectedRoute
      auth={false}
      component={
        Object {
          "$$typeof": Symbol(react.memo),
          "WrappedComponent": [Function],
          "compare": null,
          "type": [Function],
        }
      }
      path="/profile"
    />
    <ProtectedRoute
      auth={false}
      component={
        Object {
          "$$typeof": Symbol(react.memo),
          "WrappedComponent": [Function],
          "compare": null,
          "type": [Function],
        }
      }
      path="/roles"
    />
    <ProtectedRoute
      auth={false}
      component={
        Object {
          "$$typeof": Symbol(react.memo),
          "WrappedComponent": [Function],
          "compare": null,
          "type": [Function],
        }
      }
      path="/buses"
    />
    <ProtectedRoute
      auth={false}
      component={
        Object {
          "$$typeof": Symbol(react.memo),
          "WrappedComponent": [Function],
          "compare": null,
          "type": [Function],
        }
      }
      path="/simulation"
    />
    <ProtectedRoute
      auth={false}
      component={[Function]}
      path="/buses"
    />
    <ProtectedRoute
      auth={false}
      component={
        Object {
          "$$typeof": Symbol(react.memo),
          "WrappedComponent": [Function],
          "compare": null,
          "type": [Function],
        }
      }
      path="/routes"
    />
    <ProtectedRoute
      auth={false}
      component={
        Object {
          "$$typeof": Symbol(react.memo),
          "WrappedComponent": [Function],
          "compare": null,
          "type": [Function],
        }
      }
      path="/dashboard_operator"
    />
    <ProtectedRoute
      auth={false}
      component={
        Object {
          "$$typeof": Symbol(react.memo),
          "WrappedComponent": [Function],
          "compare": null,
          "type": [Function],
        }
      }
      path="/assign_drivers_buses"
    />
    <Route
      component={[Function]}
      path="/*"
    />
  </Switch>
</BrowserRouter>
`;
